src/thor/pathalgorithm.cc

```diff
@@ -0,0 +1,128 @@
+#include "thor/pathalgorithm.h"
+#include "baldr/datetime.h"
+#include "midgard/logging.h"
+#include "sif/edgelabel.h"
+
+#include <chrono>
+#include <ctime>
+
+namespace dt = valhalla::baldr::DateTime;
+namespace sc = std::chrono;
+
+namespace valhalla {
+namespace thor {
+
+// helper function to initialize the object from a location
+TimeInfo TimeInfo::make(valhalla::Location& location, baldr::GraphReader& reader) {
+  // No time to to track
+  if (!location.has_date_time()) {
+    return {false};
+  }
+
+  // Get the timezone of the first edge's end node that we can
+  int timezone_index = 0;
+  for (const auto& pe : location.path_edges()) {
+    const baldr::GraphTile* tile = nullptr;
+    const auto* edge = reader.directededge(baldr::GraphId(pe.graph_id()), tile);
+    timezone_index = reader.GetTimezone(edge->endnode(), tile);
+    if (timezone_index != 0)
+      break;
+  }
+
+  // Set the origin timezone to be the timezone at the end node use this for timezone changes
+  if (timezone_index == 0) {
+    LOG_WARN("No timezone for location using default");
+    timezone_index = 1;
```

Comment:
this is different from before. previously the code would ERROR but continue using timezone index 0 (which is invalid). that would set all of the timing information below to epoch seconds 0 and second of week 0 and so on. pretty much useless. by defaulting to a timezone the effect is that we'll at least get a local time and a second of week but the only thing that wont work will be timezone offsetting (so it will look like the world is all one big timezone). this is a better experience for someone who makes the data without timezones imho..

in the future we should force timezone inclusion in the data building process but thats a larger effort